src/map.rs

```diff
@@ -1141,21 +1148,16 @@ where
     }
 
     fn add_count(&self, n: isize, resize_hint: Option<usize>, guard: &Guard<'_>) {
-        // TODO: implement the Java CounterCell business here
-
-        use std::cmp;
-        let mut count = match n.cmp(&0) {
-            cmp::Ordering::Greater => self.count.fetch_add(n, Ordering::SeqCst) + n,
-            cmp::Ordering::Less => self.count.fetch_sub(n.abs(), Ordering::SeqCst) - n,
-            cmp::Ordering::Equal => self.count.load(Ordering::SeqCst),
-        };
+        self.counter.add(n);
```

Comment:
The optimization of dealing with `n = 0` here seems worthwhile to keep I think.

Also, separately, it feels weird that `fast-counter` holds an `isize` not a `usize`. I realize it makes the logic there a little easier (you don't need to figure out whether to do `fetch_add` or `fetch_sub`), but given that it's a _counter_ shouldn't we be okay with expecting it to always be positive?

Reply:
100%, I'll re-add the optimisation around `n = 0` my mistake in just copying over